diem-move/diem-framework/core/tests/AccountLimits.move

```diff
@@ -0,0 +1,183 @@
+    fun publish_window_non_existent_limit_address(dr: signer, tc: signer, bob: signer) {
+        setup(&dr, &tc, &bob);
+        AccountLimits::publish_window<XUS>(&dr, &bob, @0x42 /* non-exsistent */);
+    }
+
+    #[test(dr = @DiemRoot, tc = @TreasuryCompliance, bob = @0x2)]
+    #[expected_failure(abort_code = 6)]
+    fun publish_unrestricted_limits_for_testing_twice(dr: signer, tc: signer, bob: signer) {
+        setup(&dr, &tc, &bob);
+        AccountLimits::publish_unrestricted_limits_for_testing<XUS>(&bob);
+    }
+
+    #[test(dr = @DiemRoot, tc = @TreasuryCompliance, bob = @0x2)]
+    fun update_limits_definition_1(dr: signer, tc: signer, bob: signer) {
+        setup(&dr, &tc, &bob);
+        AccountLimits::update_limits_definition<XUS>(
+            &tc,
+            Signer::address_of(&bob),
+            100, /* new_max_inflow */
+            200, /* new_max_outflow */
+            150, /* new_max_holding_balance */
+            10000, /* new_time_period */
+        )
+    }
+
+    #[test(dr = @DiemRoot, tc = @TreasuryCompliance, bob = @0x2)]
+    fun update_limits_definition_2(dr: signer, tc: signer, bob: signer) {
+        setup(&dr, &tc, &bob);
+        AccountLimits::update_limits_definition<XUS>(
+            &tc,
+            Signer::address_of(&bob),
+            0, /* new_max_inflow */
+            0, /* new_max_outflow */
+            150, /* new_max_holding_balance */
+            10000, /* new_time_period */
+        )
+    }
+
+    #[test(dr = @DiemRoot, tc = @TreasuryCompliance, bob = @0x2)]
+    fun update_limits_definition_twice(dr: signer, tc: signer, bob: signer) {
+        setup(&dr, &tc, &bob);
+        AccountLimits::update_limits_definition<XUS>(
+            &tc,
+            Signer::address_of(&bob),
+            100, /* new_max_inflow */
+            200, /* new_max_outflow */
+            150, /* new_max_holding_balance */
+            10000, /* new_time_period */
+        );
+        AccountLimits::update_limits_definition<XUS>(
+            &tc,
+            Signer::address_of(&bob),
+            0, /* new_max_inflow */
+            0, /* new_max_outflow */
+            150, /* new_max_holding_balance */
+            10000, /* new_time_period */
+        )
+    }
+
+    #[test(dr = @DiemRoot, tc = @TreasuryCompliance, bob = @0x2)]
+    #[expected_failure(abort_code = 258)]
+    fun update_limits_definition_non_tc(dr: signer, tc: signer, bob: signer) {
+        setup(&dr, &tc, &bob);
+        AccountLimits::update_limits_definition<XUS>(
+            &dr,
+            Signer::address_of(&bob),
+            100, /* new_max_inflow */
+            200, /* new_max_outflow */
+            150, /* new_max_holding_balance */
+            10000, /* new_time_period */
+        )
+    }
+
+    #[test(dr = @DiemRoot, tc = @TreasuryCompliance, bob = @0x2)]
+    #[expected_failure(abort_code = 5)]
+    fun update_limits_definition_non_exsistent(dr: signer, tc: signer, bob: signer) {
```

Comment:
typo in this function name: "exsistent"

Reply:
I see the same typo in a few other places in this file, so be sure to fix them all